tasks/ceph_deploy.py

```diff
@@ -574,6 +619,87 @@ def cli_test(ctx, config):
         if config.get('rhbuild'):
             admin.run(args = ['sudo', 'yum', 'remove', 'ceph-deploy', '-y'])
 
+	    
+def get_all_pg_info(rem_site, testdir):
+    """
+    Get the results of a ceph pg dump
+    """
+    info = rem_site.run(args=[
+        'sudo',
+        'adjust-ulimits',
+        'ceph-coverage',
+        '{tdir}/archive/coverage'.format(tdir=testdir),
+        'ceph', 'pg', 'dump',
+        '--format', 'json'], stdout=StringIO())
+    all_info = json.loads(info.stdout.getvalue())
+    return all_info['pg_stats']
+
+def osd_scrub_pgs(ctx, config):
+    """
+    Scrub pgs when we exit.
+
+    First make sure all pgs are active and clean.
+    Next scrub all osds.
+    Then periodically check until all pgs have scrub time stamps that
+    indicate the last scrub completed.  Time out if no progess is made
+    here after two minutes.
+    """
+    retries = 12
+    delays = 10
+    vlist = ctx.cluster.remotes.values()
+    testdir = teuthology.get_testdir(ctx)
+    rem_site = ctx.cluster.remotes.keys()[0]
+    all_clean = False
+    for _ in range(0, retries):
+	stats = get_all_pg_info(rem_site, testdir)
+        states = [stat['state'] for stat in stats]
+        if len(set(states)) == 1 and states[0] == 'active+clean':
+            all_clean = True
+            break
+        log.info("Waiting for all osds to be active and clean.")
+        time.sleep(delays)
+    if not all_clean:
+        log.info("Scrubbing terminated -- not all pgs were active and clean.")
+        return
+    check_time_now = time.localtime()
+    time.sleep(1)
+    for slists in vlist:
+        for role in slists:
+            if role.startswith('osd.'):
+                log.info("Scrubbing osd {osd}".format(osd=role))
+                rem_site.run(args=[
+                            'sudo',
+                            'adjust-ulimits',
+                            'ceph-coverage',
+                            '{tdir}/archive/coverage'.format(tdir=testdir),
+                            'ceph', 'osd', 'deep-scrub', role])
+    prev_good = 0
+    gap_cnt = 0
+    loop = True
+    while loop:
+	stats = get_all_pg_info(rem_site, testdir)
+        timez = [stat['last_scrub_stamp'] for stat in stats]
```

Comment:
@vasukulkarni there are some inconsistencies with the indentation that are showing up in more than one spot. I believe that may be because there is a mix of tabs and spaces. 

I would either switch everything to spaces for the whole file (in vim this can be done with `:set expandtab` and then `:retab 4`), or I would go through each line that has this issue. The `cat` command can help identify which ones have tabs with the following command:

    cat -e -t tasks/ceph_deploy.py

Tabs will show up like `^I`